code/src/reducers/todos.js

```diff
@@ -0,0 +1,40 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const initialState = {
+  list: {
+    name: "To do list",
+    items: [
+      {
+        description: "What to do?",
+        done: false
+      }
+    ]
+  }
+};
+
+export const todos = createSlice({
+  name: "todos",
+  initialState: initialState,
+  reducers: {
+    addTodo: (state, action) => {
+      const { itemInfo } = action.payload;
+      state.list.items.push(itemInfo);
```

Comment:
Also add an id to your todos to be able to control them in a better way.
You can use a timestamp as you id, then you know they will always be unique.

Reply:
Hi, I tried to change this and add an id to the todos but cant get it to work. Could you give me some pointers how to change from index to id in the different files?